pyquil/operator_estimation.py

```diff
@@ -240,24 +375,68 @@ def read_json(fn):
         return json.load(f, object_hook=_operator_object_hook)
 
 
-def _local_pauli_eig_prep(op: str, idx: int):
-    """
-    Generate gate sequence to prepare a the +1 eigenstate of a Pauli operator, assuming
-    we are starting from the |00..00> ground state.
-
-    :param op: A string representation of the Pauli operator whose +1 eigenstate we'd like to
-        prepare.
-    :param idx: The index of the qubit that the preparation is acting on
-    :return: A program which will prepare the requested state.
-    """
-    if op == 'X':
-        return Program(RY(pi / 2, idx))
-    elif op == 'Y':
-        return Program(RX(-pi / 2, idx))
-    elif op == 'Z':
+def _one_q_sic_prep(index, qubit):
+    """Prepare the index-th SIC basis state."""
+    if index == 0:
         return Program()
 
-    raise ValueError(f'Unknown operation {op}')
+    theta = 2 * np.arccos(1 / np.sqrt(3))
+    zx_plane_rotation = Program([
+        RX(-pi / 2, qubit),
+        RZ(theta - pi, qubit),
+        RX(-pi / 2, qubit),
+    ])
+
+    if index == 1:
```

Comment:
If these are all mutually-exclusion branches, wouldn't it be better to use `if` and `elif`? Gives intepreter/compiler opportunity to skip the following conditionals.

Reply:
done